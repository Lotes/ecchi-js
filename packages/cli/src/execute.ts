import { spawnSync } from "node:child_process";

export function execute(fileName: string|undefined) {
  if (!fileName) {
    console.error("No filename provided");
    return;
  }
  const mainFile = require.resolve("@ecchi-js/language");
  const args = [mainFile, fileName].map((s) => s.replace(/\\/g, "/"));
  const output = spawnSync(process.execPath, args, { encoding: "utf8" });
  return output.stdout;
}
