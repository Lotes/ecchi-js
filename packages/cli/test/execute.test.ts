import { expect, test } from "vitest";
import { execute } from "../src/execute.js";
import { join } from "path";

test("execute", async () => {
  const fileName = join(__dirname, "..", "..", "..", "resources", "OneType.ecchi");
  const result = await execute(fileName);
  expect(result).toMatchSnapshot();
});