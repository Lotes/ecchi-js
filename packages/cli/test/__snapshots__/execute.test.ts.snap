// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`execute 1`] = `
"import { Reflection, Cache, LRUCache, cacheCommonExpressions, cacheSubjectExpressions, Key, merge } from "@ecchi-js/core";

interface UserType {
  $type: "UserType";
  id: number;
  name: string;
  email: string;
  createdAt: string;
  updatedAt: string;
}

export type $Types = {
  UserType: UserType,
}

export const $Reflection = new Reflection<$Types>({
  UserType: [0,  1],
});

export type $UserType = UserType;


export type $Role = ;
export type $Subject = ;
export type $Subjects = {
  
};
export type $Actions = {
  
};

export type CanOptions = {
  I: $UserType;
  actingAs?: $Role[];
  cache?: Cache<Key, any>;
} & ();

const DefaultCache = new LRUCache<Key, any>(128);

export function can({
  I: user,
  actingAs = [],
  when,
  subject,
  doWhat,
  cache = DefaultCache
}: CanOptions) {
  type Common = readonly [null];
  const commonExpressions = cacheCommonExpressions<Common>([
    (): null => null
  ] as const, [user], cache);

  const subjectHandlers: { [K in $Subject]: (subject: $Subjects[K]) => boolean} = {
    
  };
  return subjectHandlers[when](subject as any);
}
"
`;
